src/llmcompressor/transformers/compression/quantization_format.py

```diff
@@ -44,74 +94,26 @@ def infer_quantization_format(
-            return CompressionFormat.nvfp4_pack_quantized
 
-        if is_weight_only:  # w4a16 and w8a16
-            is_valid_pack = all(
-                weight_arg.num_bits in [4, 8]
-                and weight_arg.type == QuantizationType.INT.value
-                for weight_arg in weight_args
-            )
-            if not is_valid_pack:  # packing only valid for int4 and int 8
-                return CompressionFormat.naive_quantized
-            if is_24_structure:
-                for arg in weight_args:
-                    if (
-                        arg.strategy is not QuantizationStrategy.CHANNEL.value
-                        and arg.strategy is not QuantizationStrategy.GROUP.value
-                    ):
-                        # marlin24 kernel only applicable for channel/group quantization
-                        return CompressionFormat.pack_quantized
-                return CompressionFormat.marlin_24
-            return CompressionFormat.pack_quantized
-        else:  # w8a8 float and int
-            if len(weight_args) == 1:
-                if (
-                    weight_args[0].type == QuantizationType.FLOAT.value
-                    and weight_args[0].num_bits == 8
-                ):
-                    return CompressionFormat.float_quantized
-                if weight_args[0].type == QuantizationType.INT.value:
-                    return CompressionFormat.int_quantized
-
-            return CompressionFormat.naive_quantized
-    else:
-        # format will be inferred from config
+    if not save_compressed:
         return None
 
+    if quantization_format:
+        return [quantization_format]
 
-def _get_unique_quant_args(model):
-    """
-    Gets a list of all the unique quantization settings present in model
-    """
-    quant_info_weight = []
-    quant_info_inputs = []
+    unique_formats = []
     for submodule in model.modules():
         if is_module_quantized(submodule):
             weight_scheme = submodule.quantization_scheme.weights
             input_scheme = submodule.quantization_scheme.input_activations
-            if weight_scheme is not None:
-                if weight_scheme not in quant_info_weight:
-                    quant_info_weight.append(weight_scheme)
-            if input_scheme is not None:
-                if input_scheme not in quant_info_inputs:
-                    quant_info_inputs.append(input_scheme)
-
-    return quant_info_weight, quant_info_inputs
+            if weight_scheme is None:
+                continue  # no weight quant - nothing to compress
+            compression_format = _get_quant_method(
+                input_scheme, weight_scheme, sparsity_structure
+            )
+            submodule.quantization_scheme.format = compression_format.value
+            if compression_format not in unique_formats:
+                unique_formats.append(compression_format)
+    if len(unique_formats) > 0:
+        return unique_formats
+    return None
```

Comment:
Is it possible to stop supporting nullable formats? Supporting `CompresssionFormat.dense` and `None` leads to a  lot of redundant parsing logic and loose typing

Reply:
We really only support iirc. None just gets mapped to Dense.

Reply:
It's be nice if we could stop using None and always return Dense instead.

Reply:
Yeah I'll verify this is how it's being mapped when I move this function to compressed-tensors and clean some of this up.